malaria-app-android/src/main/java/com/peacecorps/malaria/notifications/receiver/DrugNotificationReceiver.java

```diff
@@ -0,0 +1,250 @@
+package com.peacecorps.malaria.notifications.receiver;
+
+import android.app.AlarmManager;
+import android.app.PendingIntent;
+import android.content.BroadcastReceiver;
+import android.content.Context;
+import android.content.Intent;
+import android.media.MediaPlayer;
+import android.net.Uri;
+import android.widget.Toast;
+
+import com.peacecorps.malaria.R;
+import com.peacecorps.malaria.code.reciever.DrugReminderCallerReceiver;
+import com.peacecorps.malaria.data.AppDataManager;
+import com.peacecorps.malaria.data.db.DbHelper;
+import com.peacecorps.malaria.notifications.service.AlarmService;
+import com.peacecorps.malaria.notifications.DrugNotificationUtils;
+import com.peacecorps.malaria.utils.CalendarFunction;
+import com.peacecorps.malaria.utils.InjectionClass;
+import com.peacecorps.malaria.utils.ToastLogSnackBarUtil;
+
+import java.util.Calendar;
+import java.util.Date;
+
+/**
+ * Created by Anamika Tripathi on 9/8/18.
+ */
+public class DrugNotificationReceiver extends BroadcastReceiver {
+    private int flag;
+    private AppDataManager dataManager;
+
+    @Override
+    public void onReceive(Context context, Intent intent) {
+
+        if (intent != null) {
+            dataManager = InjectionClass.provideDataManager(context);
+            String action = intent.getAction();
+            ToastLogSnackBarUtil.showDebugLog(action);
+            if (action != null) {
+                switch (action) {
+                    case DrugNotificationUtils.ACTION_REJECT_MEDICINE:
+                        //medicine not taken action
+                        medicineNotTaken(context);
+                        break;
+                    case DrugNotificationUtils.ACTION_SNOOZE_MEDICINE:
+                        //snooze
+                        if (flag == 0 || flag == 2) {
```

Comment:
this is not a blocker, but its good to explain what is flag 0 and flag 2 stands for?